chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,442 @@
+        """Function to preprocess inputs
+
+        Args:
+            prediction (:class:`~chainer.Variable` or \
+            :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`):
+                Prediction output
+            labels (:class:`~chainer.Variable` or \
+            :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`):
+                Ground truth output
+
+        Returns:
+            (:class:`~chainer.Variable` or \
+            :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`) :
+                Prediction output
+            (:class:`~chainer.Variable` or \
+            :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`) :
+                Ground truth output
+        """
+        # Reshape layers to prepare for processing
+
+        p_shape = prediction.shape
+        l_shape = labels.shape
+
+        prediction = reshape(prediction, shape=(
+            p_shape[0], p_shape[1], p_shape[2] * p_shape[3]))
+        labels = reshape(labels, shape=(
+            l_shape[0], l_shape[1], l_shape[2] * l_shape[3]))
+
+        return prediction, labels
+
+    def __call__(self, prediction, labels, n_objects, gt_idx):
+        """Applies discriminative margin based clustering loss
+
+        The execution steps are:
+
+        - Reshape inputs to prepare for loss calculation
+        - Calculate means
+        - Calculate variance term
+        - Calculate distance term
+        - Calculate regularization term
+        - Add weights to all and return loss value
+
+        Args:
+            prediction(:class:`~chainer.Variable` or \
+                       :class:`numpy.ndarray` or \
+                       :class:`cupy.ndarray`) :
+                       segmentation prediction output
+                       (batch size, total instance count, width, height)
+            labels(:class:`~chainer.Variable` or \
+                   :class:`numpy.ndarray` or \
+                   :class:`cupy.ndarray`) :
+                   segmentation ground truth
+                   (batch size, total instance count, width, height)
+            n_objects(:class:`~chainer.Variable` or \
+                      :class:`numpy.ndarray` or \
+                      :class:`cupy.ndarray`) :
+                      number of objects in ground truth
+                      (batch size, )
+            gt_idx(:class:`~chainer.Variable` or \
+                   :class:`numpy.ndarray` or \
+                   :class:`cupy.ndarray`) :
+                   indexes of non-zero ground truths
+                   (batch size, variable length)
+
+        Returns:
+            (:class:`~chainer.Variable` or \
+            :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`):
+                (alpha * variance_loss) +
+                (beta * distance_loss) +
+                (gamma * regularizer_loss)
+        """
+
+        buffer = chainer.config.type_check
+        chainer.config.type_check = False
```

Comment:
Sorry, I missed this code. Can't we change code while preventing the loss from disabling type checking?